File: php/examples/01_auth_oidc.php
<?php

require_once __DIR__ . '/../vendor/autoload.php';

use UIID\SDK\UIIDClient;

echo "=== 01. OIDC & Authentication Workflow (PHP) ===\n\n";

$client = new UIIDClient(clientId: 'app_550e8400', clientSecret: 'sec_f47ac10b');

// 1. Authorize URL
$url = $client->getAuthorizeUrl('https://myapp.com/callback');
echo "[OIDC] 1. Authorize URL:\n  {$url}\n\n";

echo "[OIDC] 2. Available OIDC Methods:\n";
echo " - \$client->exchangeCodeForToken(\$code, \$redirectUri)\n";
echo " - \$client->refreshToken(\$refreshToken)\n";
echo " - \$client->getUserInfo()\n";
echo " - \$client->getOidcDiscovery()\n";
echo " - \$client->getJwks()\n\n";

File: python/examples/01_auth_oidc.py
import json
from uiid_client import UIIDClient

def main():
    print("=== 01. OIDC & Authentication Workflow (Python) ===\n")

    client = UIIDClient(
        client_id="app_550e8400_client_id",
        client_secret="sec_f47ac10b_client_secret"
    )

    # 1. Authorize URL
    auth_url = client.get_authorize_url("https://myapp.com/callback")
    print("[OIDC] 1. Authorize URL:\n ", auth_url, "\n")

    print("[OIDC] 2. OIDC Discovery & JWKS Specs Available via SDK Methods:")
    print(" - client.get_oidc_discovery()")
    print(" - client.get_jwks()\n")

if __name__ == "__main__":
    main()
